packages/payload/src/config/types.ts

```diff
@@ -593,7 +598,7 @@ export type Config = {
    */
   cookiePrefix?: string
   /** Either a whitelist array of URLS to allow CORS requests from, or a wildcard string ('*') to accept incoming requests from any domain. */
-  cors?: '*' | string[]
+  cors?: '*' | CORSConfig | string[]
```

Comment:
Please also update the doc block to call out the extra configurability of CORSConfig so that it is clear that you can put the extra object configuration for origins + headers.